bin/install

```diff
@@ -11,111 +11,132 @@ install_elixir() {
     local tmp_download_dir=$TMPDIR
   fi
 
+  # running install commands in a subshell
+  # we don't want to disturb current working dir
+  if [ "$install_type" = "version" ]; then
+    (install_elixir_version $version $install_path $tmp_download_dir)
+  else
+    (install_elixir_ref $version $install_path $tmp_download_dir)
+  fi
+
+  mkdir -p $install_path/.mix/archives
+}
+
+
+install_elixir_version() {
+  local version=$1
+  local install_path=$2
+  local tmp_download_dir=$3
+
   # path to the tar file
-  local source_path=$(get_download_file_path $install_type $version $tmp_download_dir)
-  download_source_file $install_type $version $source_path
+  local source_path="$tmp_download_dir/elixir-precompiled-${version}.zip"
+  download_source_file_for_version $version $source_path
 
-  if [ "$install_type" = "version" ]
-  then
-    echo "==> Copying release into place"
-  else
-    echo "==> Making the release"
+  echo "==> Copying release into place"
+
+  if ! type "unzip" &> /dev/null; then
+    echo "ERROR: unzip not found"
+    exit 1
   fi
 
-  # running this in a subshell
-  # we don't want to disturb current working dir
-  (
-    if [ "$install_type" = "version" ]; then
-      if ! type "unzip" &> /dev/null; then
-        echo "ERROR: unzip not found"
-        exit 1
-      fi
-
-      unzip -q $source_path -d $install_path || exit 1
-    else
-      tar zxf $source_path -C $install_path --strip-components=1 || exit 1
-      cd $install_path
-      make clean test
-
-      if [ $? -ne 0 ]; then
-        rm -rf $install_path
-        exit 1
-      fi
-    fi
-
-    mkdir -p $install_path/.mix/archives
-  )
+  unzip -q $source_path -d $install_path || exit 1
 }
 
 
-download_source_file() {
-  local install_type=$1
-  local version=$2
-  local download_path=$3
-  local download_url=$(get_download_url $install_type $version)
+install_elixir_ref() {
+  local ref=$1
+  local install_path=$2
+  local tmp_download_dir=$3
+
+  # path to the tar file
+  local source_path="$tmp_download_dir/elixir-ref-${version}-src.tar.gz"
+  download_source_file_for_ref $ref $source_path
+
+  echo "==> Making the release"
+
+  tar zxf $source_path -C $install_path --strip-components=1 || exit 1
+  cd $install_path
+  make clean test
+
+  if [ $? -ne 0 ]; then
+    rm -rf $install_path
+    exit 1
+  fi
+}
+
+
+download_source_file_for_version() {
```

Comment:
Should this be named `download_zip_file_for_version` to make it clear this function isn't extracting the actual code out of the Zip file? Same question for the function below.

Reply:
Makes sense. Changed in 56e29b0.